### PR TITLE
Add: HMAC validator policy for WebSub APIs

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
- HMAC validation needs to be performed on the requests coming from the publishers if enabled. This is applicable for the WebSub APIs.

## Goals
- Gateway(or the Policy Engine) should perform the HMAC validation when a request is received from a publisher. 

## Issue
- https://github.com/wso2-enterprise/apim-saas/issues/1448

## Summary by CodeRabbit

* **New Features**
  * Added an HMAC-based request authentication policy supporting multiple algorithms with configurable header name, algorithm, secret key, and an option to allow unauthenticated requests.
  * Added a policy definition to register and validate the policy configuration.

* **Tests**
  * Added comprehensive tests covering success/failure flows, multiple algorithms, edge cases, metadata updates, and response handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: policies/hmac-validator/hmac.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+	providedAlgorithm := strings.TrimSpace(parts[0])
+	providedSignature := strings.TrimSpace(parts[1])
+
+	if providedSignature == "" {
+		return p.handleAuthFailure(ctx, 401, "json", "Valid header is required", "missing signature in authorization header")
+	}
+
+	// Validate the algorithm matches the expected algorithm
+	if !strings.EqualFold(providedAlgorithm, expectedAlgorithm) {
```
Comment (renuka-fernando) [inline policies/hmac-validator/hmac.go:114]: If we can read the algorithm from the header value, why is it read as a Policy param?

File: policies/hmac-validator/hmac_test.go [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+		{"HMAC-SHA512", "hmac-sha512"},
+		{"SHA384", "sha384"},
+		{"HMAC-SHA384", "hmac-sha384"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Generate valid HMAC signature
+			signature := generateTestHMAC(t, testBody, testSecretKey, tc.algorithm)
+			headerValue := tc.algorithm + "=" + signature
```
Comment (renuka-fernando) [inline policies/hmac-validator/hmac_test.go:83]: We should use some hard coded signature in the test right?

Comment (renuka-fernando) [reply policies/hmac-validator/hmac_test.go:83]: Otherwise, we just use the same function for test and we use the same func to generate the testcase
